Controller/ProfilerController.php

```diff
@@ -117,4 +119,15 @@ private function explainOtherPlatform(Connection $connection, $query)
         return $connection->executeQuery('EXPLAIN ' . $query['sql'], $params, $query['types'])
             ->fetchAll(PDO::FETCH_ASSOC);
     }
+
+    /**
+     * @param mixed[] $query
+     */
+    private function explainOraclePlatform(Connection $connection, array $query)
+    {
+        $connection->executeQuery('EXPLAIN PLAN FOR ' . $query['sql']);
```

Comment:
Why isn't this using `SET statement_id = ` like oracle manual specifies? Doesn't your current implementation pull out records for every statement recorded?

Reply:
I did not know about the `SET statement_id = ` variant, but `DBMS_XPLAN.DISPLAY` without parameters will return the plan for the most recent explained query.

Reply:
I have investigated further: The results of `EXPLAIN PLAN` are inserted into a table. Thus they are properly isolated on the transaction level.

Reply:
[From Ask Tom](https://asktom.oracle.com/pls/apex/f?p=100:11:0::::P11_QUESTION_ID:9535917800346105419) 
> Note that plan_table is a synonym to sys.plan_table$ (unless you've created your own...). This is a session-level temporary table (on commit preserve rows). So your data is wiped from this when you close the session regardless of whether you commit or rollback.

So another sessions can't interfere with "our" explain plan.